packages/design-toolkit/src/components/flashcard/index.tsx

```diff
@@ -0,0 +1,202 @@
+
+  return (
+    <FlashcardContext.Provider value={{ isLoading }}>
+      <div {...rest} className={container({ className })}>
+        {children}
+      </div>
+    </FlashcardContext.Provider>
+  );
+}
+Flashcard.displayName = 'Flashcard';
+
+export function FlashcardHero(props: FlashcardComponentProps) {
+  const { children, className, ...rest } = props;
+  const { isLoading } = useContext(FlashcardContext);
+
+  if (isLoading) {
+    return (
+      <div {...rest} className={hero({ className })}>
+        <Skeleton className={skeleton()} />
+        <Skeleton className={skeleton({ className: 'max-w-1/2' })} />
+      </div>
+    );
+  }
+
+  return (
+    <div {...rest} className={hero()}>
+      {children}
+    </div>
+  );
+}
+FlashcardHero.displayName = 'FlashcardHero';
+
+export function FlashcardIdentifier(props: FlashcardComponentProps) {
+  const { children, className, ...rest } = props;
+
+  return (
+    <div {...rest} className={identifier({ className })}>
+      {children}
+    </div>
+  );
+}
+FlashcardIdentifier.displayName = 'FlashcardIdentifier';
+
+export function FlashcardHeader(props: FlashcardComponentProps) {
+  const { children, className, ...rest } = props;
+
+  return (
+    <div {...rest} className={header({ className })}>
+      {children}
+    </div>
+  );
+}
+FlashcardHeader.displayName = 'FlashcardHeader';
+
+export function FlashcardSubheader(props: FlashcardComponentProps) {
+  const { children, className, ...rest } = props;
+
+  return (
+    <div {...rest} className={subHeader({ className })}>
+      {children}
+    </div>
+  );
+}
+FlashcardSubheader.displayName = 'FlashcardSubheader';
+
+export function FlashcardSecondary(props: FlashcardComponentProps) {
+  const { children, className, ...rest } = props;
+
+  return (
+    <div {...rest} className={secondaryContainer({ className })}>
+      {children}
+    </div>
+  );
+}
+FlashcardSecondary.displayName = 'FlashcardSecondary';
+
+export function FlashcardSecondaryData(props: FlashcardComponentProps) {
+  const { children, className, ...rest } = props;
+  const { isLoading } = useContext(FlashcardContext);
+
+  // While loading, don't display.
+  if (isLoading) {
+    return null;
+  }
+
+  return (
+    <h2 {...rest} className={secondaryData({ className })}>
+      {children}
+    </h2>
+  );
+}
+FlashcardSecondaryData.displayName = 'FlashcardSecondaryData';
+
+export function FlashcardDetailsContainer(
+  props: FlashcardDetailContainerProps,
+) {
+  const { details, classNames, ...rest } = props;
+  const { isLoading } = useContext(FlashcardContext);
+
+  if (!details?.length) {
+    return null;
+  }
+
+  const displayDetails = details.slice(0, 5);
+
+  return (
+    <div
+      {...rest}
+      className={detailsContainer({ className: classNames?.container })}
+    >
+      {/* Limit to the first 5 items in details array.*/}
+      {displayDetails.map((item) => {
+        const detailId = uuid();
```

Comment:
I know my previous comment led to this, and this is a little funky to create new uuids for these on every render. It'll lead to unnecessary re-renders.

A more stable approach would be to require an id in the item data shape. The id should be created when the `details` data is created, not when it's rendered.